src/C/sparse.c

```diff
@@ -3161,7 +3146,45 @@ spmatrix_setitem_ij(spmatrix *A, int_t i, int_t j, number *value) {
 
   SP_ROW(A)[k] = i;
   write_num[SP_ID(A)](SP_VAL(A), k, value, 0);
+
+  return 0;
 }
+static int spmatrix_additem_ij(spmatrix *A, int_t i, int_t j, number *value) {
+    /*
+     * spmatrix_additem_ij
+     * @author: Nichlos West, CURENT
+     */
+    number val;
+
+    if (OUT_RNG(i, SP_NROWS(A)) || OUT_RNG(j, SP_NCOLS(A))) {
+        PY_ERR_INT(PyExc_IndexError, "index out of range");
+    }
+
+    i = CWRAP(i, SP_NROWS(A));
+    j = CWRAP(j, SP_NCOLS(A));
+
+    if (!(spmatrix_getitem_ij(A, i, j, &val) || realloc_ccs(A->obj, SP_NNZ(A)+1))) {
```

Comment:
Hi @cuihantao 

Thanks for your contribution. I also had a function like this one in my power flow program, but it will be nice to have it inside K/CVOXPT

 if spmatrix_getiem_if returns >= , we don't need to realloc since we already have that entry in SP_NNZ. Let me try to propose you an alternative to this.

Reply:
Sounds great. Thanks for your reply. I believe you have much more experience with CVXOPT's C APIs.

Looking forward to having it merged so that my package can depend solely on KVXOPT. 

By the way, do you consider publishing in on conda-forge? That will at least solve the macOS library issue.